pandas/tests/test_resample.py

```diff
@@ -2192,6 +2192,17 @@ def test_resample_datetime_values(self):
         res = df['timestamp'].resample('2D').first()
         tm.assert_series_equal(res, exp)
 
+    def test_apply_to_empty_series(self):
+        # GH 14313
+        series = self.create_series()[:0]
+
+        for freq in ['M', 'D', 'H']:
+            result = series.resample(freq).apply(lambda x: 1)
+            expected = series.resample(freq).apply(np.sum)
+
+            assert result.name == expected.name
```

Comment:
@jreback 

I've already tried to move this to `Base`.But I got the failure on `TestPeriodIndex`
```
pandas/core/resample.py:316: in aggregate
    self._set_binner()
pandas/core/resample.py:244: in _set_binner
    self.binner, self.grouper = self._get_binner()
pandas/core/resample.py:252: in _get_binner
    binner, bins, binlabels = self._get_binner_for_time()
pandas/core/resample.py:695: in _get_binner_for_time
    return self.groupby._get_time_bins(self.ax)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pandas.core.resample.TimeGrouper object at 0x7f8ba410d860>, ax = PeriodIndex([], dtype='period[D]', freq='D')

    def _get_time_bins(self, ax):
        if not isinstance(ax, DatetimeIndex):
            raise TypeError('axis must be a DatetimeIndex, but got '
>                           'an instance of %r' % type(ax).__name__)
E           TypeError: axis must be a DatetimeIndex, but got an instance of 'PeriodIndex'
```

I didn't find in tests whether `PeriodIndex` is testing with `apply`.
Is this correct behavior? I mean the test failure on `PeriodIndex`